daemons/fenced/pacemaker-fenced.c

```diff
@@ -788,12 +776,29 @@ fencer_metadata(void)
                                "Pacemaker's fence daemon, formerly known as "
                                "stonithd");
 
-    char *s = pcmk__format_option_metadata(name, desc_short, desc_long,
-                                           pcmk__opt_context_none,
-                                           fencer_options,
-                                           PCMK__NELEM(fencer_options));
-    printf("%s", s);
-    free(s);
+    pcmk__output_t *tmp_out = NULL;
+    xmlNode *top = NULL;
+    const xmlNode *metadata = NULL;
+    char *metadata_s = NULL;
+
+    int rc = pcmk__output_new(&tmp_out, "xml", "/dev/null", NULL);
+    if (rc != pcmk_rc_ok) {
+        return rc;
+    }
+
+    out->message(tmp_out, "option-list", name, desc_short, desc_long,
+                 pcmk__opt_none, fencer_options, true);
```

Comment:
I ran this PR through my formatted output arg checking, and aside from me having to change something, this was the one line that causes problems:
```pacemaker-fenced.c:789:5: error: Expected 'uint32_t', but got 'int' in argument 6
  789 |     out->message(tmp_out, "option-list", name, desc_short, desc_long,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  790 |                  pcmk__opt_none, fencer_options, true);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Casting `pcmk__opt_none` to `uint32_t` fixes this.  I'm not sure I want to do anything about this in my tool since gcc expects an unsigned type but thinks it's getting a signed one.

Reply:
That would be reasonable to fix on the caller side. In `pcmk__daemon_metadata()`, I already cast an `enum pcmk__opt_flags` to `uint32_t`.

Reply:
I did change this `out->message` call, but in hindsight it's pretty nonstandard to do so. We usually **don't** cast our enum values to `uint32_t` before passing them as flag group arguments.

Reply:
Defining `pcmk__opt_none` to `0U` should take care of it (or `UINT32_C(0)`, but I don't think that's necessary in this case)

Reply:
Good idea, I'll do that as well as `(1U << x)` for the other flags